pkg/telemetry/phonehome/gatherer.go

```diff
@@ -92,18 +91,16 @@ func (g *gatherer) identify() {
 }
 
 func (g *gatherer) loop() {
-	defer g.procs.Done()
-
 	// Send initial data on start:
-	g.procs.Add(1)
 	g.identify()
-	ticker := time.NewTicker(g.period)
+	ticker := g.tickerFactory(g.period)
 	defer ticker.Stop()
 	for !g.stopSig.IsDone() {
 		select {
-		case <-ticker.C:
-			g.procs.Add(1)
-			go g.identify()
+		case _, ok := <-ticker.C:
+			if ok {
+				go g.identify()
+			}
```

Comment:
Indeed, the extra goroutine may only happen if we close the channel before we Stop in the tests. I decided to add the condition just to not rely on the order, and because anyway it doesn't make sense to gather when the ticker is stopped.